libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2245,6 +2245,22 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 		}
 	}
 
+	if (_functionType->kind() == FunctionType::Kind::BytesConcat)
+	{
+		solAssert(paramArgMap.empty(), "");
```

Comment:
Maybe assert that it is a positional call?

Reply:
Added